x-pack/plugins/security_solution/server/lib/detection_engine/rule_execution_log/event_log_adapter/event_log_client.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IEventLogger, IEventLogService } from '../../../../../../event_log/server';
+import { RuleExecutionStatus } from '../../../../../common/detection_engine/schemas/common/schemas';
+import { LogStatusChangeArgs } from '../types';
+import { RuleExecutionLogAction, RULE_EXECUTION_LOG_PROVIDER } from './constants';
+
+const statusSeverityDict: Record<RuleExecutionStatus, number> = {
+  [RuleExecutionStatus.succeeded]: 0,
+  [RuleExecutionStatus['going to run']]: 10,
+  [RuleExecutionStatus.warning]: 20,
+  [RuleExecutionStatus['partial failure']]: 20,
+  [RuleExecutionStatus.failed]: 30,
+};
+
+interface FindExecutionLogArgs {
+  ruleIds: string[];
+  spaceId: string;
+  logsCount?: number;
+  statuses?: RuleExecutionStatus[];
+}
+
+interface LogExecutionMetricsArgs {
+  ruleId: string;
+  spaceId: string;
+  metrics: EventLogExecutionMetrics;
+}
+
+interface EventLogExecutionMetrics {
+  searchDurationSum?: number;
+  indexingDurationSum?: number;
+  executionGapDuration?: number;
+}
+
+interface IEventLogClient {
+  find: (args: FindExecutionLogArgs) => Promise<{}>;
+  logStatusChange: (args: LogStatusChangeArgs) => Promise<void>;
+  logExecutionMetrics: (args: LogExecutionMetricsArgs) => Promise<void>;
+}
+
+export class EventLogClient implements IEventLogClient {
+  private sequence = 0;
+  private eventLogger: IEventLogger;
+
+  constructor(eventLog: IEventLogService) {
+    this.eventLogger = eventLog.getLogger({
+      event: { provider: RULE_EXECUTION_LOG_PROVIDER },
+    });
+  }
+
+  public async find({ ruleIds, spaceId, statuses, logsCount = 1 }: FindExecutionLogArgs) {
+    return {}; // TODO implement
+  }
+
+  public async logExecutionMetrics({
+    ruleId,
+    metrics,
+    spaceId, // TODO write spaceId
```

Comment:
Is there anything that prevents writing it?

Reply:
If we're passing it in, we should write it to the new `space_ids` field (as an array!) - I don't see that new event log field being used anywhere!  And yes, it's nice to have :-)

Reply:
RBAC section of the RFC is not finished yet, but it looks like we'll need the `space_ids` field anyway. Passed it to the log.